src/main.rs

```diff
@@ -135,7 +136,7 @@ fn setup(args: &[String]) -> Setup {
     let device_id = librespot::session::device_id(&name);
 
     let cache = matches.opt_str("c").map(|cache_location| {
-        Cache::new(PathBuf::from(cache_location))
+        Cache::new(PathBuf::from(cache_location), !matches.opt_present("disable-audio-cache"))
```

Comment:
Just a nit, could you extract `!matches.opt_present("disable-audio-cache")` to a separate variable ?